modules/apps/site/site-cms-site-initializer/src/main/resources/META-INF/resources/js/main_view/props_transformer/utils/displayUndoDeleteSuccessToast.ts

```diff
@@ -0,0 +1,76 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2025 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import {OpenToastProps, openToast} from 'frontend-js-components-web';
+import {sub} from 'frontend-js-web';
+
+import {
+	getFormattedLabel,
+	getFormattedLink,
+} from '../../../common/utils/getFormattedText';
+import {restoreItemAction} from '../actions/restoreItemAction';
+
+const recycleBinToastInfo = {
+	className: 'recycle-bin-link',
+	linkName: Liferay.Language.get('recycle-bin'),
+	url: '/cms/recycle-bin',
+};
+
+export default function displayUndoDeleteSuccessToast(
+	label: string,
+	loadData: () => {},
+	method: string,
+	restoreURL: string
+) {
+	const {className, linkName, url} = recycleBinToastInfo;
+
+	const formmatedRecycleBinLink = getFormattedLink(
+		className,
+		linkName,
+		Liferay.ThemeDisplay.getPathFriendlyURLPublic() + url
+	);
+	const formattedItemLabel = getFormattedLabel(label);
+
+	const openToastSuccessProps: OpenToastProps = {
+		message: sub(
+			Liferay.Language.get('x-was-moved-to-the-x'),
+			formattedItemLabel,
+			formmatedRecycleBinLink
+		),
+		type: 'success',
+	};
+
+	const undoButtonLabel = Liferay.Language.get('undo');
+	const undoButtonClassName = 'cms-delete-item-undo-button';
+
+	openToastSuccessProps.message =
+		openToastSuccessProps.message +
+		`<div class="alert-footer">
+				<div class="btn-group" role="group">
+					<button class="btn btn-sm btn-success alert-btn ${undoButtonClassName}">${undoButtonLabel}</button>
+				</div>
+		</div>`;
+
+	async function handleOnClick() {
+		await restoreItemAction(label, loadData, method, restoreURL);
+	}
+
+	openToastSuccessProps.onClick = ({event, onClose: closeToast}) => {
+		const {target} = event;
+
+		if (
+			target instanceof HTMLElement &&
+			target.classList.contains(undoButtonClassName)
+		) {
+			handleOnClick();
+
+			// @ts-ignore
```

Comment:
I don't know why this [CloseEvent](https://developer.mozilla.org/en-US/docs/Web/API/CloseEvent) is [required](https://github.com/liferay/liferay-portal/blob/242a3dec6e0e4e8067659aa2ed939e4600bb3aac/modules/apps/frontend-js/frontend-js-components-web/src/main/resources/META-INF/resources/toast/openToast.tsx#L95)

From what I see, the event type is meant for WebSockets, so I’m not sure what event it’s supposed to handle here , it actually looks like a bad typing.

For now, I think the `// ts-ignore` is fine, but maybe we should check with FI if this `CloseEvent` is expected.